gui/wxpython/main_window/notebook.py

```diff
@@ -0,0 +1,141 @@
+"""
+@package frame.notebook
+
+@brief Custom AuiNotebook class and class for undocked AuiNotebook frame
+
+Classes:
+ - notebook::MapPageFrame
+ - notebook::MapNotebook
+
+(C) 2022 by the GRASS Development Team
+
+This program is free software under the GNU General Public License
+(>=v2). Read the file COPYING that comes with GRASS for details.
+
+@author Linda Kladivova <lindakladivova gmail.com>
+@author Anna Petrasova <kratochanna gmail.com>
+"""
+
+import wx
+import wx.lib.agw.aui as aui
+
+from gui_core.wrap import SimpleTabArt
+
+
+class MapPageFrame(wx.Frame):
+    """Frame for independent map display window."""
+
+    def __init__(self, parent, mapdisplay, title):
+        wx.Frame.__init__(self, parent=parent, title=title)
+        self.mapdisplay = mapdisplay
+        self.SetSize(mapdisplay.GetSize())
+        self.sizer = wx.BoxSizer(wx.HORIZONTAL)
+        self.SetSizerAndFit(self.sizer)
+        self.Bind(wx.EVT_CLOSE, self.OnClose)
+
+    def closeFrameNoEvent(self):
+        """Close frame without generating OnClose event."""
+        self.Unbind(wx.EVT_CLOSE)
+        self.Close()
+        self.Bind(wx.EVT_CLOSE, self.OnClose)
+
+    def OnClose(self, event):
+        """Close frame and associated layer notebook page."""
+        self.mapdisplay.OnCloseWindow(event=None, askIfSaveWorkspace=True)
+
+
+class MapNotebook(aui.AuiNotebook):
+    """Map notebook class. Overrides some AuiNotebook classes.
+    Takes into consideration the dock/undock functionality.
+    """
+
+    def __init__(
+        self,
+        parent,
+        agwStyle=aui.AUI_NB_DEFAULT_STYLE | aui.AUI_NB_TAB_EXTERNAL_MOVE | wx.NO_BORDER,
+    ):
+        self.parent = parent
+        super().__init__(parent=self.parent, id=wx.ID_ANY, agwStyle=agwStyle)
+        client_size = self.parent.GetClientSize()
+        self.SetPosition(wx.Point(client_size.x, client_size.y))
+        self.SetSize(430, 200)
```

Comment:
Is this really needed?

Reply:
you are right, it was there before this PR but it is not needed